python/lsst/daf/butler/core/dimensions/_universe.py

```diff
@@ -78,59 +88,53 @@ class DimensionUniverse:
     For internal use only.
     """
 
-    def __new__(cls, config: Optional[Config] = None) -> DimensionUniverse:
-        # Normalize the config and apply defaults.
-        config = DimensionConfig(config)
-
-        # First see if an equivalent instance already exists.
-        version = config["version"]
+    def __new__(
+        cls,
+        config: Optional[Config] = None, *,
+        version: Optional[int] = None,
+        builder: Optional[DimensionConstructionBuilder] = None,
+    ) -> DimensionUniverse:
+        # Try to get a version first, to look for existing instances; try to
+        # do as little work as possible at this stage.
+        if version is None:
+            if builder is None:
+                from ._config import DimensionConfig
```

Comment:
Why do you import here, is there import cycle somewhere?

Reply:
There was a cycle in an earlier version. I managed to flatten that one out but forgot to change this; I'll do that now.

Reply:
👍 There is another instance of this import below, just in case.